### DISCUSSION TITLE
How to set the config in aioble

### DISCUSSION THREAD

**Discussion by brianreinhold** [micropython/micropython#13585] [Libraries & Drivers] [answered]
This should be easy. I want to set the bond flag to True. aioble.core has this method
```
def config(*args, **kwargs):
    ensure_active()
    return ble.config(*args, **kwargs)
```
However, when I call `config(bond=True)` I get an 'Unknown Config Param' error

I have even tried
```
    ble: bluetooth.BLE = bluetooth.BLE()
    ble.active(True)
    ble.config(bond=True)
```
and I get the same error. The documentation for both aioble and the low level define 'bond' as one of the parameters or key words.

Anyone know what I am doing wrong? I am trying to solve the problem where the library responds to a security request with the bond flag cleared which no health device accepts.

I have come to the conclusion there is some bug in the low-level bluetooth implementation and perhaps its port to the PICO-W. The BLE.config method gives the same error which means the aioble library will also report that error.

**Comment by glenn20** [comment, marked as answer]
The `bond` config parameter is only enabled if micropython is compiled with the `MICROPY_PY_BLUETOOTH_ENABLE_PAIRING_BONDING` compilation flag enabled, which it would appear is not enabled by default on your port. Unfortunately, I dont know if this can be enabled for the pico-w by building your own micropython image.

See https://github.com/micropython/micropython/blob/ed15b3c6c67f4626dde1ac4281978b3bef1c72ce/extmod/modbluetooth.h#L65 and https://github.com/micropython/micropython/blob/ed15b3c6c67f4626dde1ac4281978b3bef1c72ce/extmod/modbluetooth.c#L399.

**Reply by brianreinhold** [reply]
Yes, I found out that its not a bug in the stack, but the fact the pairing/bonding option is disabled for the PICO-W. I wish I knew how the project works so I could fix it, but I only have knowledge of Bluetooth, not the micropython project. It seems funny that it would be disabled without a good reason for it being disabled, but given that the C-SDK for the PICO-W does support bonding/pairing, I do not understand that reasoning!

Looking at those files for MICROPY_PY_BLUETOOTH_ENABLE_PAIRING_BONDING to be defined also requires MICROPY_PY_BLUETOOTH_USE_SYNC_EVENTS to be defined. It also looks like MICROPY_PY_BLUETOOTH_ENABLE_CENTRAL_MODE is disabled (it's grayed out in the VS Code editor in any case. That is not very good either. 

Do you know where these are determined?

**Reply by brianreinhold** [reply]
On the PICO-W setting these parameters will set the flag correctly but pairing/bonding will not work. The implementation for pairing is not complete and it is going to take a considerable effort to complete it.
